wayland-client/src/rust_imp/display.rs

```diff
@@ -21,6 +23,11 @@ pub(crate) struct DisplayInner {
 
 impl DisplayInner {
     pub unsafe fn from_fd(fd: RawFd) -> Result<Arc<DisplayInner>, ConnectError> {
+        if std::env::var_os("WAYLAND_DEBUG").is_some() {
```

Comment:
In order to match the semantics of `libwayland`, we should activate the debugging only if `WAYLAND_DEBUG` is set to either `client` or `1` (an symmetrically, activate it server-side only if it is set to either `server` or `1`).